packages/http/src/validator/validators/__tests__/body.spec.ts

```diff
@@ -45,5 +46,57 @@ describe('HttpBodyValidator', () => {
         ).toMatchSnapshot();
       });
     });
+
+    describe('body is form-urlencoded with deep object style', () => {
+      it('returns no validation errors', () => {
+        expect(
+          httpBodyValidator.validate(
+            encodeURI(
+              'key[a]=str&key[b][ba]=str&key[b][bc][0]=bc0&key[b][bc][1]=bc1&key[c][0][a]=c0a&key[c][0][b]=c0b&key[c][1][a]=c0a&key[c][1][b]=c0b&other=other',
+            ),
+            [
+              {
+                mediaType: 'application/x-www-form-urlencoded',
+                encodings: [{ property: 'key', style: HttpParamStyles.DeepObject }],
+                schema: {
+                  type: 'object',
+                  properties: {
+                    key: {
+                      type: 'object',
+                      properties: {
+                        a: { type: 'string' },
+                        b: {
+                          type: 'object',
+                          properties: {
+                            ba: { type: 'string' },
+                            bb: { type: 'object' },
+                            bc: { type: 'array' },
+                          },
+                          required: ['ba', 'bb', 'bc'],
+                        },
+                        c: {
+                          type: 'array',
+                          items: {
+                            type: 'object',
+                            properties: {
+                              a: { type: 'string' },
+                              b: { type: 'string' },
+                            },
+                            required: ['a', 'b'],
+                          },
+                        },
+                      },
+                      required: ['a', 'b', 'c'],
+                    },
+                  },
+                  required: ['key'],
+                },
+              },
+            ],
+            'application/x-www-form-urlencoded',
+          ),
+        ).toEqual([]);
+      });
+    });
```

Comment:
Maybe we could 
1. add a failing test
2. Break down this super test into a set of sub tests? (nested, array…)

Reply:
1. added failing test
2. I simplified the test to only ensure that deserializer works properly